docs/testing-operators.md

```diff
@@ -1,120 +1,404 @@
+
+
+Create the following as the file `operator-group.yaml`
+
+```
+apiVersion: operators.coreos.com/v1alpha2
+kind: OperatorGroup
+metadata:
+  name: my-operatorgroup
+  namespace: default
+spec:
+  targetNamespaces:
+  - default
+```
+
+Deploy the `OperatorGroup` resource:
+
+```
+kubectl apply -f operator-group.yaml
+```
+
+### 8. Create a Subscription
+
+The last piece ties together all of the previous steps. A `Subscription` is created to the operator. Save the following to a file named: `operator-subscription.yaml`:
+
+```
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: my-operator-subsription
+  namespace: default
+spec:
+  channel: <channel-name>
+  name: my-operator
+  source: johndoe-operators
+  sourceNamespace: olm
+```
+
+### 9. Verify Operator health
+
+Watch your Operator being deployed by OLM from the catalog source created by Operator Marketplace with the following command:
+
+```
+kubectl get clusterserviceversion -n default
+
+NAME                 DISPLAY       VERSION   REPLACES   PHASE
+my-operator.v1.0.0   My Operator   1.0.0                Succeeded
+```
+
+> The above command assumes you have created the `Subscription` in the `default` namespace. Adjust accordingly if you have selected a different namespace.
+
+
+If your Operator deployment (CSV) shows a `Succeeded` in the `InstallPhase` status, your Operator is deployed successfully. If that's not the case check the `ClusterServiceVersion` objects status for details.
+
+Optional also check your Operator's deployment:
+
+```
+kubectl get deployment -n default
+```
+
+## Testing Operator Deployment on OpenShift
+
+On OpenShift Container Platform and OKD 4.1 or newer `operator-marketplace` and `operator-lifeycle-manager` are already installed. You can start right away by creating an `OperatorSource` in the `openshift-marketplace` namespace as a user with the `cluster-admin` role.
+
+### 1. Create the OperatorSource
+
+An `OperatorSource` object is used to define the external datastore we are using to store operator bundles. More information including example can be found in the documentation included in the `operator-marketplace` [repository](https://github.com/operator-framework/operator-marketplace#operatorsource).
+
+**Replace** `johndoe` in `metadata.name` and `spec.registryNamespace` with your quay.io username in the example below and save it to a file called `operator-source.yaml`.
+
+```
+apiVersion: operators.coreos.com/v1
+kind: OperatorSource
+metadata:
+  name: johndoe-operators
+  namespace: openshift-marketplace
+spec:
+  type: appregistry
+  endpoint: https://quay.io/cnr
+  registryNamespace: johndoe
+  displayName: "John Doe's Operators"
+  publisher: "John Doe"
+```
+
+Create the object:
+
+```
+kubectl apply -f operator-source.yaml
```

Comment:
Doesn't it make more sense to use the `oc` cli when running against openshift? Sure both will work, but if someone is coming here specifically to deploy with openshift it would probably make more sense to use the native client.